### ISSUE TITLE
Better handeling of not a git repository

### ISSUE THREAD

**Issue by mjarkk** [jesseduffield/lazygit#447]
**Is your feature request related to a problem? Please describe.**
When someone types lazygit in a not git repo it craches

**Describe the solution you'd like**
First of all crashing isn't the right behaviour because nothing is broken, the user is just in a wrong dir without the `.git` folder.  

But what i personally would like to see is a popup with the recently visited git repos where i can select a repo to use in lazygit.  
Beside that it would be nice if it's possible change the current work dir from the opened repo.  

That way it's easy to go to a recently opened git repo from anywhere on the system and when someone would like to make changes to that repo they can just press something like `shift + Q` to quit and change there current working dir to lazygit's git repo dir.  

Often when i work on projects i have multiple terminal tabs open in multiple git repos but to get these tabs open after a system bootup it takes a stupid long time. When having the above in lazygit this can be solved because lazygit already knows all my visited repos the `cd ../../../someGitServer/user/project` can be replaced with opening lazygit, select a recently opened git repo and do a quit + change the current dir.  

**Describe alternatives you've considered**
- **CD into a project.**  
Yes most people will have no problem with that and i agree but when having to deal with for example multiple micro services over multiple git repo's from multiple origins this gets annoying very quickly 

- **Use a smarter shell like fish/zsh**  
From the things i have used to make chaining long dirs easier this is one of the best solutions for changing the dir fast, fish has a nice feather where it suggests the path for you and it works oke but it still often suggest the half path because because i'm to lazy to type the full path every time like: (where i meant to directly go further `~/go/src/github.com/mjarkk/somerepo`)![image](https://user-images.githubusercontent.com/15320763/57676013-d15d5b00-7623-11e9-84a8-3f0633b17b2a.png)
 

**Additional context**
It would be great if it's also possible to change to another git repo from lazygit when inside a git repo  
If that's possible i can just have Lazygit open in a separate tab and quickly push changes from multiple git repos

**Comment by jesseduffield** [comment]
I have thought about this exact same thing myself. Firstly, currently we can change to recent repos by going to the status panel and hitting 's', so potentially it wouldn't be too hard to just re-use that code to provide that functionality at startup if we're not in a git repo already. It might be worth having a different GUI for it that simply renders the list, or we can use the existing GUI struct but we'll need to ensure it doesn't break when we're not currently in a git repo.

Secondly, I agree that it would be great if the shell's directory could be changed by lazygit so that when you exit it, you're in the same repo and you don't end up entering bash commands in the wrong context. But I believe that is impossible from lazygit because a subshell cannot change the directory of it's parent shell. Having said that, this stack overflow thread https://stackoverflow.com/questions/13753157/bash-script-to-change-parent-shell-directory suggests to use a function in your ~/.bashrc that would call the program, and then use its output to change directory afterwards.

so maybe it would be something like

```
lg () {
  output=$(path/to/lazygit)

  cd $output
}
```

We'd need to add some extra stuff into that function to ensure the output isn't an error or something, but I think that could work. Then we could maybe add another keybinding within lazygit that adds that function to the user's ~/.bashrc file.

What are your thoughts?

**Comment by mjarkk** [comment]
That's annoying we can't change the working dir from lazygit,  
But how about opening a new shell from lazygit that way we can actually set the work and when lazygit is called from the shell inside lazygit it just closes the shell and put the user back inside of lazygit.  

If we would do that it's even possible to use lazygit as start commando for a new terminal tab.  

But this also has few problems, for example opening an extra terminal tab will continue from where the first shell is if lazygit isn't the original commando.    

Both solutions seem a bit off, messing in the shell config to get a program "properly working" doesn't sound right to me but executing a shell from lazygit doesn't also looks a bit wrong.  
Though now i think of it's actually used quite often for example when i want to use shell functions that are different in fish i run bash from the fish shell

**Comment by jesseduffield** [comment]
I agree that both situations are awkward, although I believe that the approach of adding something to the user's ~/.bashrc is the less awkward, because you'll get the same behaviour regardless of where you open lazygit, and you're not missing out on features that e.g. your terminal program has specifically for when you're entering top-level commands.

We do have a half-way solution in that you can run a command from within lazygit using shift+X in the files panel, which is run from the directory of the opened repo, but this won't suffice for all use cases. 

As it stands, I believe the ~/.bashrc function may be the only way to do this.

**Comment by jesseduffield** [comment]
worth noting that I came across another TUI today which gets you to add a function to your rc file to change directory on exit: https://github.com/jarun/nnn

**Comment by mjarkk** [comment]
There solusion seems fine to me,  

it doesn't look to complicated and it doesn't have terminal output on exit, they just place a file in the temp dir with the user's there current dir and on exit they cd to the content of the file.
